ocaml/xapi/message_forwarding.ml

```diff
@@ -2671,23 +2673,26 @@ module Forward = functor(Local: Custom_actions.CUSTOM_ACTIONS) -> struct
 			info "Pool_update.destroy: pool update = '%s'" (pool_update_uuid ~__context self);
 			Local.Pool_update.destroy ~__context ~self
 
-		let attach ~__context ~self ~host =
-			info "Pool_update.attach: pool update = '%s' host = '%s'" (pool_update_uuid ~__context self) (host_uuid ~__context host);
-			let local_fn = Local.Pool_update.attach ~self ~host in
-			do_op_on ~local_fn ~__context ~host
-				(fun session_id rpc -> Client.Pool_update.attach rpc session_id self host)
+		let attach ~__context ~self =
+			info "Pool_update.attach: pool update = '%s'" (pool_update_uuid ~__context self);
+			let local_fn = Local.Pool_update.attach ~self in
+			let vdi = Db.Pool_update.get_vdi ~__context ~self in
+			do_op_on ~local_fn ~__context ~host:(Xapi_pool_update.get_candidate_host ~__context ~vdi)
```

Comment:
get_candidate_host will always return master if share SR.
Not sure forward_vdi_op have the same behavior? Simply read the code seems not guaranteed.